include/evm_runtime/transaction.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <optional>
+#include <eosio/eosio.hpp>
+#include <evm_runtime/types.hpp>
+#include <silkworm/core/rlp/encode.hpp>
+#include <silkworm/core/types/transaction.hpp>
+
+namespace evm_runtime {
+
+using silkworm::Bytes;
+using silkworm::ByteView;
+
+struct transaction {
+
+  transaction() = delete;
+  explicit transaction(bytes& rlptx) : rlptx_(std::move(rlptx)) {}
+  explicit transaction(silkworm::Transaction& tx) : tx_(std::move(tx)) {}
+
+  const bytes& get_rlptx() {
+    if(!rlptx_) {
+      eosio::check(tx_.has_value(), "no tx");
+      Bytes rlp;
+      silkworm::rlp::encode(rlp, tx_.value());
+      rlptx_.emplace(bytes{rlp.begin(), rlp.end()});
+    }
+    return rlptx_.value();
+  }
+
+  const silkworm::Transaction& get_tx() {
+    if(!tx_) {
+      eosio::check(rlptx_.has_value(), "no rlptx");
+      ByteView bv{(const uint8_t*)rlptx_->data(), rlptx_->size()};
+      silkworm::Transaction tmp;
+      eosio::check(silkworm::rlp::decode(bv, tmp) && bv.empty(), "unable to decode transaction");
+      tx_.emplace(tmp);
+    }
+    return tx_.value();
+  }
+
+private:
+  std::optional<bytes>  rlptx_;
+  std::optional<silkworm::Transaction> tx_;
```

Comment:
I think these should both be `mutable` and `get_rlptx` and `get_tx` should be const to indicate through the interface that it is not actually meaningfully mutating the state represented by this transaction wrapper.